filters/auth/tokeninfocache.go

```diff
@@ -0,0 +1,145 @@
+package auth
+
+import (
+	"container/list"
+	"sync"
+	"time"
+
+	"github.com/zalando/skipper/filters"
+)
+
+type (
+	tokeninfoCache struct {
+		client tokeninfoClient
+		size   int
+		ttl    time.Duration
+		now    func() time.Time
+
+		mu    sync.Mutex
```

Comment:
Did you check if RWMutex would be helpful?

Reply:
I did not, somehow I have an impression that we should not prefer it to sync.Mutex but I can not find a link why.

Reply:
I am not sure if the mutex would be a bottleneck here, the benchmark with high number of goroutines does not look too off from the low number:
```
BenchmarkTokeninfoCache/tokens=100,cacheSize=100,p=0-8           1506963               811.9 ns/op           344 B/op          3 allocs/op
BenchmarkTokeninfoCache/tokens=100,cacheSize=100,p=10000-8       1802880               655.4 ns/op           365 B/op          3 allocs/op
```
Also cached() method checks expiration so if we use RWMutex it would need to re-acquire write lock somehow which would complicate things.

Reply:
true